File: app/src/main/java/com/lazycouple/restapiclient/ui/viewModel/RestRequestViewModel.java
package com.lazycouple.restapiclient.ui.viewModel;

import android.databinding.Bindable;

import com.android.databinding.library.baseAdapters.BR;
import com.lazycouple.restapiclient.ui.data.CustomResponse;
import com.lazycouple.restapiclient.db.model.Parameter;
import com.lazycouple.restapiclient.ui.presenter.RestRequestPresenter.Method;

/**
 * Created by amuyu on 2017. 6. 12..
 */

public class RestRequestViewModel extends AdapterViewModel<Parameter> {

    String url;
    Method method = Method.GET;
    boolean isRequest;

    // response
    CustomResponse response;
    String code;
    String body;


    @Bindable
    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
        notifyPropertyChanged(BR.url);
    }

    public Method getMethod() {
        return method;
    }

    public void setMethod(Method method) {
        this.method = method;
        notifyPropertyChanged(BR.methodText);
    }

    public CustomResponse getResponse() {
        return response;
    }

    public void setResponse(CustomResponse response) {
        this.response = response;
        notifyPropertyChanged(BR.code);
        notifyPropertyChanged(BR.body);
    }

    @Bindable
    public String getMethodText() {
        return method.name();
    }

    @Bindable
    public String getCode() {
        if(response != null) return String.valueOf(response.code);
        return "";
    }

    @Bindable
    public String getBody() {
        if(response != null) {
            if(response.code == 200) return response.body;
            else return response.errorbody;
        }
        return "";
    }

    @Bindable
    public boolean isRequest() {
        return isRequest;
    }

    public void setRequest(boolean request) {
        isRequest = request;
        notifyPropertyChanged(BR.request);
    }
}

File: app/src/main/kotlin/com/lazycouple/restapiclient/ui/viewModel/RestRequestViewModel.kt
package com.lazycouple.restapiclient.ui.viewModel

import android.databinding.Bindable

import com.android.databinding.library.baseAdapters.BR
import com.lazycouple.restapiclient.repository.local.model.Parameter
import com.lazycouple.restapiclient.ui.data.CustomResponse
import com.lazycouple.restapiclient.ui.presenter.RestRequestPresenter.Method

/**
 * Created by amuyu on 2017. 6. 12..
 */

class RestRequestViewModel : AdapterViewModel<Parameter>() {

    internal var url: String? = null
    internal var method = Method.GET
    internal var isRequest: Boolean = false

    // response
    internal var response: CustomResponse? = null
    internal var code: String? = null
    internal var body: String? = null


    @Bindable
    fun getUrl(): String? {
        return url
    }

    fun setUrl(url: String) {
        this.url = url
        notifyPropertyChanged(BR.url)
    }

    fun getMethod(): Method {
        return method
    }

    fun setMethod(method: Method) {
        this.method = method
        notifyPropertyChanged(BR.methodText)
    }

    fun getResponse(): CustomResponse? {
        return response
    }

    fun setResponse(response: CustomResponse) {
        this.response = response
        notifyPropertyChanged(BR.code)
        notifyPropertyChanged(BR.body)
    }

    val methodText: String
        @Bindable
        get() = method.name

    @Bindable
    fun getCode(): String {
        if (response != null) return response!!.code.toString()
        return ""
    }

    @Bindable
    fun getBody(): String? {
        if (response != null) {
            if (response!!.code == 200)
                return response!!.body
            else
                return response!!.errorbody
        }
        return ""
    }

    @Bindable
    fun isRequest(): Boolean {
        return isRequest
    }

    fun setRequest(request: Boolean) {
        isRequest = request
        notifyPropertyChanged(BR.request)
    }
}

